snmp/datadog_checks/snmp/config.py

```diff
@@ -32,13 +32,23 @@ def __init__(self, instance, warning, global_metrics, mibs_path, profiles):
         self.transport = self.get_transport_target(instance, timeout, retries)
 
         self.ip_address = instance['ip_address']
+
+        if not self.metrics and not profiles_by_oid:
+            raise ConfigurationError('Metrics list must contain at least one metric')
```

Comment:
(nit) The error message seems not taking into account the second part of the condition `not profiles_by_oid`.